### PR TITLE
option callable-types set to falseπ

### PR DESCRIPTION
**Pull request by jakagacic**
I suggest setting option callable-types to false, so we can extract function calls to interfaces, instead of having to use types, an example can be found here:

[https://www.typescriptlang.org/docs/handbook/interfaces.html#function-types](https://www.typescriptlang.org/docs/handbook/interfaces.html#function-types)

rule: [https://palantir.github.io/tslint/rules/callable-types/](https://palantir.github.io/tslint/rules/callable-types/)

### CODE REVIEW

**Review by vilimco — APPROVED** [review]
I approve of this change. 

If you define a callable interface, you can extend it at some point (it is not often practice at all, but in TS it is a nice possibility). On the other hand, callable `type` does not have that option. And `type` defining reminds me of older versions of TS when everything was defined with `type`.

**Comment by krukru** [comment]
Can you give me an example that will trigger an error for this rule?

**Comment by vilimco** [comment]
```js
export interface CallableInterface {
    () => number;
}
```

will trigger tslint and ask for a refactor to this:

```js
export type CallableInterface = () => number
```

**Comment by krukru** [comment]
Seems reasonable, but why not give the lambda expression a name and turn it into a proper method signature?

**Comment by vilimco** [comment]
Then you will need to pass an object with method, this way you provide lambdas directly, you just have a proper type for it.

While I agree that it is also good approach, this is most often used approach in TS/JS where you provide callbacks for stuff when you are not using Promises.

**Review by krukru — APPROVED** [review]
I did some checking and it seems reasonable to enable this. Lets get one more approve before we merge.

**Comment by jakagacic** [comment]
@ilesar @bonzzy ping

**Comment by krukru** [comment]
Since we never got the third and final review, the assumption is that they agree and that is why the review is missing.
@jakagacic you can merge this and publish to npm

**Pull request merged**
